### PR TITLE
Allow user override of RGB_DISABLE_WHEN_USB_SUSPENDED for BM40 keyboard

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Allow user override to the `RGB_DISABLE_WHEN_USB_SUSPENDED` setting, based on @tzarc's suggestion in #12290

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

**Comment by filterpaper** [comment]
@tzarc Can you help add the appropriate label to avoid re-flagging while awaiting other reviewers? Thanks.

**Comment by fauxpark** [comment]
See https://github.com/qmk/qmk_firmware/pull/13146

I think the correct way to override this define now (in develop) is to just `#undef` it if it's defined, and `#define` it if it's not.

**Review by drashna — APPROVED** [review]

File: keyboards/bm40hsrgb/config.h [hunk 1]
```diff
@@ -50,6 +50,6 @@
 #ifdef RGB_DI_PIN
     #define RGB_MATRIX_KEYPRESSES // reacts to keypresses
 #endif
-#ifdef RGB_MATRIX_ENABLE
-#    define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
+#ifndef RGB_DISABLE_WHEN_USB_SUSPENDED
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```
Comment (drashna) [inline keyboards/bm40hsrgb/config.h:54]: For now: 
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```

Comment (drashna) [inline keyboards/bm40hsrgb/config.h:54]:
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```

**Comment by drashna** [comment]
Thanks!

**Pull request merged**